docs/hello_nextflow/06_hello_config.md

```diff
@@ -259,71 +555,614 @@ NXF_CHDIR=/home/gitpod/work/34/850fe31af0eb62a0eb1643ed77b84f
+
+### 4.4. Run again with the modified configuration
+
+Let's run the workflow again with the modified configuration and with the reporting flag turned on, but notice we're giving the report a different name so we can differentiate them.
+
+```bash
+nextflow run main.nf -profile my_laptop -with-report report-config-2.html
+```
+
+Once again, you probably won't notice a substantial difference in runtime, because this is such a small workload and the tools spend more time in ancillary tasks than in performing the 'real' work.
+
+However, the second report shows that our resource utilization is more balanced now, and the runtime of the `GATK_JOINTGENOTYPING` process has been cut in half.
+We probably didn't need to go all the way to 8 CPUs, but since there's only one call to that process, it's not a huge drain.
+
+**TODO: screenshots?**
+
+As you can see, this approach is useful when your processes have different resource requirements. It empowers you to can right-size the resource allocations you set up for each process based on actual data, not guesswork.
+
+!!!note
+
+    This is just a tiny taster of what you can do to optimize your use of resources.
+    Nextflow itself has some really neat [dynamic retry logic](https://training.nextflow.io/basic_training/debugging/#dynamic-resources-allocation) built in to retry jobs that fail due to resource limitations.
+    Additionally, the Seqera Platform offers AI-driven tooling for optimizing your resource allocations automatically as well.
+
+    We'll cover both of those approaches in an upcoming part of this training course.
+
+That being said, there may be some constraints on what you can (or must) allocate depending on what computing executor and compute infrastructure you're using. For example, your cluster may require you to stay within certain limits that don't apply when you're running elsewhere.
+
+### 4.5. Add resource limits to an HPC profile
+
+You can use the `resourceLimits` directive to set the relevant limitations. The syntax looks like this when it's by itself in a process block:
+
+```groovy title="Syntax"
+process {
+    resourceLimits = [
+        memory: 750.GB,
+        cpus: 200,
+        time: 30.d
+    ]
+}
+```
+
+Let's add this to the `univ_hpc` profile we set up earlier.
+
+_Before:_
+
+```groovy title="nextflow.config"
+    univ_hpc {
+        process.executor = 'slurm'
+        conda.enabled = true
+    }
+```
+
+_After:_
 
 ```groovy title="nextflow.config"
+    univ_hpc {
+        process.executor = 'slurm'
+        conda.enabled = true
+        process.resourceLimits = [
+            memory: 750.GB,
+            cpus: 200,
+            time: 30.d
+        ]
+    }
+```
+
+We can't test this since we don't have a live connection to Slurm in the Gitpod environment, but you can run it and look up the `sbatch` command in the `.command.run` script file if you'd like to see how these directives are translated into job parameters for the executor.
```

Comment:
So where can people look to see how these instructions are provided to Slurm?

Reply:
Ohh I get it now. Because Nextflow isn't *telling* Slurm what the limits are, it's *enforcing* them before submitting jobs through `sbatch`, is that right? 

You can probably tell I've only run on cloud myself, not HPC 😅

Anyway, will edit. Thanks.

Reply:
That makes sense, thank you!

Reply:
Rephrased like this, lmk if that's still wrong.

> We can't test this since we don't have a live connection to Slurm in the Gitpod environment.
> However, you can try running the workflow with resource allocations that exceed these limits, then look up the `sbatch` command in the `.command.run` script file to see what requests actually get sent to the executor.